lib/elastic_record/index/documents.rb

```diff
@@ -84,7 +84,11 @@ def index_document(id, document, doctype: model.doctype, parent: nil, index_name
           path = "/#{index_name}/#{doctype.name}/#{id}"
           path << "?parent=#{parent}" if parent
 
-          connection.json_put path, document
+          if id
+            connection.json_put path, document
+          else
+            connection.json_post path, document
+          end
```

Comment:
The model should not be required to implement the `id` method. Use `#try(:id)`.

Reply:
I think what you intended with this was that the model shouldn't need to define `primary_key`.  The change above is on `index_document` which takes `id` as it's first argument.

Reply:
That code was not shown, so I commented here.

Use try(:id). The new `record_primary_key` is useless, since that's what rails already does with `id`.

Reply:
Ah, I get it.